### PR TITLE
create API method to get current configuration of the service

### LINKED ISSUE #1967: create API method to get current configuration of the service
- [ ] Create method - getCurrentConfigurationOfAService
- [ ] URI - GET /xpanse/service/current/config/{serviceId}

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #1967
image

### CODE REVIEW

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java [hunk 1]
```diff
@@ -51,6 +52,20 @@ public class ServiceConfigurationApi {
     @Resource
     private ServiceConfigurationManager serviceConfigurationManager;
 
+    @Tag(name = "ServiceConfiguration",
+            description = "APIs for managing service's configuration.")
```
Comment (Alice1319) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java:56]: LGTM

**Comment by Alice1319** [comment]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java [hunk 2]
```diff
@@ -51,6 +52,20 @@ public class ServiceConfigurationApi {
...
+    @GetMapping(value = "/service/current/config/{serviceId}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(description = "Query the service's configuration by serviceId.")
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java:60]: Itis better to update the description with 'Query the service's  current configuration by id of the deployed service'

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java [hunk 3]
```diff
@@ -51,6 +52,33 @@ public class ServiceConfigurationApi {
...
+     */
+    @Tag(name = "ServiceConfiguration",
+            description = "APIs for managing service's configuration.")
+    @GetMapping(value = "/service/current/config/{serviceId}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(description = "Query the service's current configuration by"
+            + " id of the deployed service.")
+    @AuditApiRequest(methodName = "getCspFromServiceId")
+    public ServiceConfigurationEntity getCurrentConfigurationOfService(
```
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceConfigurationApi.java:69]: we must not return the entity as it is. If we do so, the deploy service entity is also returned.  This must be suppressed.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**